### PR TITLE
First pass standalone NFS

### PR DESCRIPTION
**Pull request by datamattsson**
For testing, replace the git clone URL with this branch.

### CODE REVIEW

File: docs/csi_driver/standalone_nfs.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# Standalone NFS Server
+
+In certain situations is desirable to run the NFS Server Provisioner image without the dual `PersistentVolumeClaim` (PVC) semantic in a more static fashion on top of a `PVC` provisioned by a non-HPE CSI Driver `StorageClass`.
+
+[TOC]
+
+## Limitations
+
+- The standalone NFS server is not part of the HPE CSI Driver and should be considered a standalone Kubernetes application altogether. The HPE CSI Driver NFS Server Provisioner NFS servers may co-exist on the same cluster and `Namespace` without risk of conflict but not recommended.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:9]:
```suggestion
- The standalone NFS server is not part of the HPE CSI Driver and is deployed as a standalone Kubernetes application. The HPE CSI Driver NFS Server Provisioner and standalone NFS server instances may co-exist on the same cluster and `Namespace` without risk of conflict but is not recommended.
```

File: docs/csi_driver/standalone_nfs.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+- The [Pod Monitor](monitor.md) which normally monitors `Pods` status for the "NodeLost" condition is not included with the standalone NFS server and recovery is at the mercy of the underlying storage platform and driver.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:10]:
```suggestion
- The [Pod Monitor](monitor.md) which normally monitors `Pods` status for "NodeLost" conditions is not included with the standalone NFS server and recovery is the responsibility of the underlying storage platform and driver.
```

File: docs/csi_driver/standalone_nfs.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+- Support is limited on the standalone NFS server and only available to select users.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:11]: Support by whom? HPE?

Who are the "select users"? Certain HPE customers?

File: docs/csi_driver/standalone_nfs.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+## Prerequisites
+
+It's assumed during the creation steps that a Kubernetes cluster is available with enough permissions to deploy privileged containers with `SYS_ADMIN` and `DAC_READ_SEARCH` capabilities. All steps are run in a terminal with `kubectl` and `git` in in the path.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:15]:
```suggestion
It's assumed the user deploying the standalone NFS server has appropriate permissions to deploy privileged containers with `SYS_ADMIN` and `DAC_READ_SEARCH` capabilities on the respective Kubernetes cluster. All steps are run in a terminal with `kubectl` and `git` in in the path.
```

File: docs/csi_driver/standalone_nfs.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+4 directories, 9 files
+```
+
+!!! important 
+    The current directory is now the "home" for the remainder of this guide.
+
+## Create an NFS Server
+
+Copy the "example" overlay into a new directory. In the examples "my-server" is used.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:57]:
```suggestion
Copy the "example" overlay into a new directory. For this guide, the folder "my-server" is used.
```

File: docs/csi_driver/standalone_nfs.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+PERSISTENCE_SIZE=16Gi
+
+# Default resource limits for the NFS server
+NFS_SERVER_CPU_LIMIT=1
+NFS_SERVER_MEMORY_LIMIT=2Gi
+```
+
+The "CLUSTER_NODE_DOMAIN_NAME" variable refers to the DNS domain name that the worker node is resolvable in, not the Kubernetes cluster DNS.
+
+The "PERSISTENCE_SIZE" is the backend `PVC` size expressed in the same format accepted by a `PVC`.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:81]: "the same format", do we have the acceptible SI units documented somewhere for reference? (M, G, T)

File: docs/csi_driver/standalone_nfs.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+service/my-server-hpe-nfs   ClusterIP   10.100.200.11   <none>        49000/TCP,2049/TCP,2049/UDP,32803/TCP,32803/UDP,20048/TCP,20048/UDP,111/TCP,111/UDP,662/TCP,662/UDP,875/TCP,875/UDP   59s
+
+NAME                                      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+persistentvolumeclaim/my-server-hpe-nfs   Bound    pvc-ae943116-d0af-4696-8b1b-1dcf4316bdc2   18Gi       RWO            vsphere-sc     58s
+
+NAME                                READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/my-server-hpe-nfs   1/1     1            1           59s
+```
+
+Make a note of the IP address assigned to "service/my-server-hpe-nfs", that is the IP address needed to mount the NFS export.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:123]:
```suggestion
Make a note of the IP address assigned to "service/my-server-hpe-nfs", that is the IP address that will be used to mount NFS export(s).
```

File: docs/csi_driver/standalone_nfs.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+!!! tip
+    If the Kubernetes cluster DNS service is resolvable from the worker node host OS, it possible to use the cluster DNS address to mount the `Service`, in this example that would be "my-server-hpe-nfs.default.svc.cluster.local".
+
+## Mounting the NFS Server
+
+There are two ways to mount the NFS server.
+
+1. Inline declaration of where to find the NFS server and NFS Export
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:132]:
```suggestion
1. Inline declaration of the NFS server and NFS Export
```
Is "Export" a proper name?

File: docs/csi_driver/standalone_nfs.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+1. Statically creating a `PersistentVolume` with the NFS server details and mount options and manually claiming the `PV` with a `PVC` using the `.spec.volumeName` parameter
+
+### Inline Declaration
+
+This is the most elegant solution as it does not require any intermediary `PVC` or `PV` and directly refers to the NFS server within a workload stanza.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:137]:
```suggestion
This is the most elegant solution as it does not require any intermediary `PVC` or `PV` and defines the NFS server within a workload stanza.
```

File: docs/csi_driver/standalone_nfs.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+
+This is an example from a `StatefulSet` workload controller having multiple replicas.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:139]:
```suggestion
This is an example of a `StatefulSet` workload controller with multiple replicas.
```

File: docs/csi_driver/standalone_nfs.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+
+Refer to the [official Kubernetes documentation](https://kubernetes.io/docs/concepts/storage/volumes/#nfs) for the built-in NFS client on how to perform static provisioning of NFS `PVs` and `PVCs`.
+
+## Expand PVC
+
+If the `StorageClass` and underlying CSI driver supports volume expansion, simply edit `overlays/my-server/environment.properties` with the new (larger) size and issue `kubectl apply -k overlays/my-server` to expand the volume.
+
+## Deleting the NFS Server
+
+Ensure no workloads have active mounts against the NFS server `Service`. If there are, those `Pods` will be stuck indefinitely.
```
Comment (c-snell) [inline docs/csi_driver/standalone_nfs.md:173]:
```suggestion
Ensure there are no workloads with active mounts to the NFS server `Service`. If the mounts are deleted while `Pods` are still active, those `Pods` will be stuck indefinitely. 
```

**Review by c-snell — CHANGES_REQUESTED** [review]
Some grammatical fixes required.

**Review by dileepds — APPROVED** [review]

**Pull request merged**